### PR TITLE
Deprecate config.yaml in testgrid

### LINKED ISSUE #13541: shard testgrid config
**What should be cleaned up or changed**:
Similar to how we have sharded up job configs by owning SIG to enable better self-service, let's shard up testgrid configs by owning SIG.

**Provide any links for context**:
Work has already started on this, linking relevant PR's:
- https://github.com/kubernetes/test-infra/pull/13472 (initial impl)
- https://github.com/kubernetes/test-infra/pull/13530 (revert)
- https://github.com/kubernetes/test-infra/pull/13539 (bugfix)

/area testgrid
/area config
/milestone v1.16
/priority important-soon
/assign @chases2

### PR DESCRIPTION
**Pull request by chases2**
Improves readme documentation to contain info that was in `config.yaml`

Includes a deprecation notice in `config.yaml`

This is part of an effort to shard testgrid's config, and should close #13541

/cc @michelle192837 @Katharine @spiffxp

### CODE REVIEW

File: config/testgrids/README.md [hunk 1]
```diff
@@ -12,10 +12,28 @@ no changes are necessary here unless you are adding a brand new dashboard.
...
+     append a new testgroup under `test_groups`, and specify the name and where to get the log.
+1.   Append a new `dashboard_tab` under the dashboard you would like to add the testgroup to,
+     or create a new `dashboard` and assign the testgroup to the dashboard.
+     * The testgroup name from a dashboard tab should match the name from a testgroup
+     * Note that a testgroup can be within multiple dashboards.
+1.   Test your new config (`bazel test //config/tests/...`)
+
+
+NOTE: If you're adding a periodic or postsubmit and don't want to specially configure your test
+group, you don't need to: [Configurator](/testgrid/cmd/configurator) implicitly assume a testgroup
```
Comment (michelle192837) [inline config/testgrids/README.md]: nit: assumes

File: config/testgrids/README.md [hunk 2]
```diff
@@ -12,10 +12,28 @@ no changes are necessary here unless you are adding a brand new dashboard.
...
+exists for all periodics and postsubmits. You still need to add presubmits here, and all jobs still
```
Comment (Katharine) [inline config/testgrids/README.md]: This is incorrect - you can and probably should just add an appropriate annotation to force it to be generated.

**Pull request merged**